src/main/kotlin/io/github/gradlenexus/publishplugin/NexusHost.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import java.net.URI
+
+enum class NexusHost(private val baseUrl: String) {
```

Comment:
Let's make this a regular data class and just have `oss` and `s01` constants in `NexusRepositoryContainer` so it's open for extension.

Reply:
I'd make this an `fun interface` or normal `class` instead. There's no need for any of the data class feature overhead.

Reply:
Then it makes sense to add a method like `nexus(name: String, urlPrefix: String)` to `NexusRepositoryContainer`.
It would enable users to create repositories (e.g. non-sonatype ones) with their own URLs. On the positive side, they won't have to manually override both `releaseUrl` and `snapshotUrl`. WDYT?

Reply:
While we're at it. Should we just provide a string with the `baseUrl` instead? @marcphilipp @vlsi

Reply:
`sonatype(host: EnumLikeType, ...)` should better have enum-like parameter since currently we know only two well-known hosts: `oss.sonatype....` and `s01....`

On the other hand, a separate method might be helpful for creating nexus server with `String`-based `baseUrl` so users can add their own servers.

`sonatype("https://nexus.internal.example.com")` would look weird in the build script. What would `sonatype` even mean there?

Reply:
Now I do get your point regarding the `nexus` function. I think we're getting off track here. `NexusHost` would be a good naming if we want to configure any nexus. Here we only want to configure the `SonatypeNexusHost` or `SonatypeHost`, so maybe we should keep the enum as is. Maybe deprecate the implicit `sonatype` factory and add the suggested `nexus` factory as a semantic differentiation.

    sonatype(oss) { ... }
    sonatype(s01) { ... }

    nexus("name") { ... }

Do you know if Nexus is fully configurable or does it provide static paths?